client/src/Components/CommonComponents/OfferCard/index.jsx

```diff
@@ -1,6 +1,137 @@
 import React from 'react';
-import UnderConstruction from '../../UnderConstruction';
+import { Card, Button, Spinner } from 'react-bootstrap';
+import PropTypes from 'prop-types';
+import './style.css';
+import { withRouter } from 'react-router-dom';
 
-export default function OfferCard() {
-  return <UnderConstruction />;
+class OfferCard extends React.Component {
+  state = {
+    offer: null,
+    statusLabel: '',
+    statusDiv: '',
+    hovered: '',
+    saved: false,
+  };
+
+  componentDidMount() {
+    const { offer, status } = this.props;
+    if (offer.saved) this.setState({ saved: true });
+    this.setState({ offer });
```

Comment:
I think it's better to do one `this.setState` here instead of three so we prevent unnecessary rendering
something like:
```jsx
componentDidMount() {
    const statusColor = {
      completed: 'green',
      finished: 'red',
      pending: 'blue',
    };
    const { offer, status } = this.props;
    this.setState(() => {
      return {
        offer,
        saved: offer.saved,
        statusLabel: `offer-card__status--${status}`,
        statusDiv: `offer-card__border--${statusColor[status]}`,
      };
    });
}
```